src/components/NFTArticle/SlateEditor/Elements/RenderElements.tsx

```diff
@@ -45,6 +45,19 @@ function EditableElementWrapper({
       at: target
     })
     setShowAddBlock(false)
+    // in order to retrieve the DOMNode and restore
+    // the selection correctly, we have to wait
+    setTimeout(() => {
+      const node = Node.get(editor, target)
+      const domNode = ReactEditor.toDOMNode(editor, node)
+      function setSelectionToNewBlock() {
+	ReactEditor.focus(editor)
+	const path = ReactEditor.findPath(editor, node)
+	const [lastLeaf, lastLeafPath] = Node.last(editor, path);
+	Transforms.select(editor, lastLeafPath)
+      }
+      domNode.tabIndex > -1 ? domNode.focus?.() : setSelectionToNewBlock()
+    })
```

Comment:
After a block is added, we check if the created element is focusable. If `true` we call the elements `focus()` method. If `false` we restore the selection by setting the focus to the editor and finding the deepest leaf path of the created node and set the selection there.